dlib/test/dnn.cpp

```diff
@@ -3703,6 +3729,7 @@ namespace
             test_softmax_all();
             test_sigmoid();
             test_mish();
+            test_leaky_relu();
```

Comment:
Add a `test_layer()` test for this new layer over in `test_layers()`.

Reply:
Good catch, I forgot to implement the inplace case for the cuda backend...